### PR TITLE
30 validate codesamples

### PR DESCRIPTION
**Pull request by madziaf**
created TestConverter and AtaruError

### CODE REVIEW

File: lib/ataru/application.rb [hunk 1]
```diff
@@ -5,6 +5,18 @@ def self.run_test_for_file(file_name)
     kramdown_doc = MarkdownLoader.load_file(file_name)
     #pulling an array of codespans/code samples from markdown document
     code_samples = Traverser.codespans(kramdown_doc)
+
+    code_samples = code_samples.map do |sample|
+      p sample
```
Comment (schultyy) [inline lib/ataru/application.rb:10]: Is the debug output here on purpose? If not, then it should be removed :)

**Comment by nanoparsec** [comment]
solves issues #30

File: lib/ataru/test_converter.rb [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+
+    if code_sample == nil
+      raise ArgumentError
+    end
+
+    if code_sample.size == 0
+      return [:warning, code_sample]
+    end
+
+    before_match = code_sample.match(/(.*) #=>/)
```
Comment (moonglum) [inline lib/ataru/test_converter.rb:13]: Yay!

**Comment by moonglum** [comment]
Nice :+1:

**Pull request merged**